src/omv/ports/stm32/sensor.c

```diff
@@ -1110,25 +1115,62 @@ void DCMI_VsyncExtiCallback()
     }
 }
 
-// To make the user experience better we automatically shrink the size of the MAIN_FB() to fit
-// within the RAM we have onboard the system.
-static void sensor_check_buffsize()
+static uint32_t get_src_bytes_per_pixel()
 {
-    uint32_t size = framebuffer_get_buffer_size();
-    uint32_t bpp;
+    switch (sensor.pixformat) {
+        case PIXFORMAT_GRAYSCALE:
+            return sensor.gs_bpp;
+        case PIXFORMAT_RGB565:
+        case PIXFORMAT_YUV422:
+            return sizeof(uint16_t);
```

Comment:
```suggestion
            return 2;
```

Reply:
I prefer the sizeof(uint16_t). It infers intent of the pixel size.

Reply:
Fixed